tasks/push-snapshot/push-snapshot.yaml

```diff
@@ -64,6 +64,35 @@ spec:
           fi
         }
 
+        translate_tags () { # Expected arguments are [tags, timestamp, git sha, 8 character git sha, source sha]
+            if [[ $1 == '' ]] ; then
+                echo ''
+                return
+            fi
+
+            SUPPORTED_VARIABLES='[
+                {"{{timestamp}}": "'$2'"},
+                {"{{ timestamp }}": "'$2'"},
+                {"{{git_sha}}": "'$3'"},
+                {"{{ git_sha }}": "'$3'"},
+                {"{{git_short_sha}}": "'$4'"},
+                {"{{ git_short_sha }}": "'$4'"},
+                {"{{source_sha}}": "'$5'"},
+                {"{{ source_sha }}": "'$5'"}
+            ]'
+            tags=$1
+
+            NUM_VARIABLES=$(jq 'length' <<< "${SUPPORTED_VARIABLES}")
+            for ((i = 0; i < $NUM_VARIABLES; i++)); do
+                variable=$(jq -c --argjson i "$i" '.[$i]' <<< "${SUPPORTED_VARIABLES}")
+                KEY=$(jq -r 'to_entries[] | .key' <<< $variable)
+                VALUE=$(jq -r 'to_entries[] | .value' <<< $variable)
+                tags=$(echo -n $tags | sed "s/$KEY/$VALUE/g")
```

Comment:
I'm wondering if we should use Jinja for this. But maybe that would be an overkill and also we might then need some check to only allow what we want to allow. So maybe this is ok for now. I was just worried that it seems a bit fragile to hardcode the whole expression like this - I think people would generally assume that it's jinja-like templating, so whitespaces inside `{{}}` shouldn't matter, but here if you accidentally do `{{ timestamp}}` it won't work. Maybe it's acceptable, but wondering what you think about this.

Reply:
Is `{{ timestamp}}` even allowed with normal jinja? I don't think this is necessarily the best way to do it, it is just a way that works and is pretty easy to understand IMO. We can always change it later

Reply:
I think that in Jinja, any whitespaces are ignored and even `{{timestamp}}` will work (but I haven't checked), but it's just more readable with the spaces. But yeah, I agree it should be ok for now.